R/axis.R

```diff
@@ -76,9 +77,25 @@ axis.gsplot <- function(object, ..., n.minor=0, tcl.minor=0.15, reverse=NULL) {
   
   for(side in sides){
     # append the side and give it defaults if it doesn't exist
+    
     object <- modify_side(object, args = list(), side=side) 
+    which.axis <- which(names(object[[as.side_name(side)]])== 'axis')
+    if (append){
+      last.axis <- tail(which.axis, 1) # get the last one
+      object[[as.side_name(side)]] <- append(object[[as.side_name(side)]], 
+                                             list('axis' = set_args('axis', side=side, package='graphics')), 
+                                             after = last.axis)
+      object[[as.side_name(side)]][[last.axis+1]] <- append_replace(object[[as.side_name(side)]][[last.axis+1]], user_args[[fun.name]])
+    } else {
+      # remove
+      if (length(which.axis) > 1){
+        # remove all axis functions other than the first one
+        object[[as.side_name(side)]] <- object[[as.side_name(side)]][-which.axis[!which.axis %in% head(which.axis, 1)]]  
```

Comment:
I would think in that example the last axis call replaces everything above it since it does not have append. 

```r
gs <- gsplot() %>% 
    points(0:1, 0:1) %>% 
    axis(side=1, at=c(0.5, 1)) %>% 
    axis(side=1, at=c(0.33))
gs
```

Creates
![image](https://cloud.githubusercontent.com/assets/13220910/19779843/7033d62c-9c48-11e6-8169-0507e82283dd.png)